src/fairseq2/recipes/utils/sweep_tag.py

```diff
@@ -244,39 +244,127 @@ def get_sweep_keys(extra_sweep_keys: Set[Hashable] | None) -> Set[Hashable]:
 @cache
 def get_default_sweep_keys() -> Set[Hashable]:
     return {
-        "batch_shuffle_window",
-        "betas",
-        "data_parallelism",
+        # Common configuration
+        "name",
+        "family",
+        "config",
+        "text_tokenizer",
+        # Top level keys
+        "model",
         "dataset",
-        "dtype",
-        "example_shuffle_window",
-        "final_lr_ratio",
-        "final_lr_scale",
-        "fp16_loss_scale",
-        "fsdp_reshard_after_forward",
-        "fsdp_wrap_granularity",
-        "gradient_accumulation",
-        "label_smoothing",
-        "lr",
-        "lr_stage_ratios",
-        "max_gradient_norm",
-        "max_num_elements",
-        "max_num_steps",
-        "max_num_tokens",
+        "gang",
+        "trainer",
+        "criterion",
+        "optimizer",
+        # Model configuration
+        "arch",
+        "checkpoint",
+        # Dataset configuration
+        "path",
+        "min_seq_len",
         "max_seq_len",
-        "mixed_precision",
-        "model",
-        "model_arch",
-        "model_config",
-        "num_lr_warmup_steps",
-        "pretrained_model",
-        "seed",
-        "split",
-        "start_lr",
-        "start_lr_scale",
-        "tensor_parallel_size",
-        "tokenizer",
+        "max_num_tokens",
+        "batch_size",
+        "example_shuffle_window",
+        "batch_shuffle_window",
+        "num_prefetch",
         "train_split",
         "valid_split",
+        # Gang configuration
+        "tensor_parallel_size",
+        "timeout",
+        "high_priority",
+        "monitored",
+        # Trainer configuration
+        "dtype",
+        "data_parallelism",
+        "fsdp",
+        "version",
+        "granularity",
+        "hybrid",
+        "reshard_after_forward",
+        "fp32_reduce",
+        "mixed_precision",
+        "gradient_accumulation",
+        "activation_checkpointing",
+        "max_gradient_norm",
+        "fp16_loss_scale",
+        "torch_compile",
+        "gc_every_n_steps",
+        "profile",
+        "gradient_check",
+        "anomaly_detection",
+        # Criterion configuration
+        "reference_model",
+        "reference_dtype",
+        "beta",
+        "nll_scale",
+        "length_normalization",
+        # Optimizer configuration
+        "lr",
+        "betas",
+        "eps",
         "weight_decay",
+        "amsgrad",
+        "maximize",
+        "capturable",
+        "differentiable",
+        "impl",
+        "use_fp32",
+        # Learning rate scheduler configuration
+        "cycle_len",
+        "num_warmup_steps",
+        "cycle_mul",
+        "lr_mul",
+        "start_lr",
+        "final_lr",
+        "final_lr_scale",
+        # Regime configuration
+        "num_steps",
+        "num_data_epochs",
+        "score_metric",
+        "lower_score_better",
+        "validate_after_n_steps",
```

Comment:
No blocker since you are on leave. Just a note though that some of the newly added keys are not meant to be used in sweep names. For instance the `xxx_after_n_steps` and `xxx_every_n_steps` options do not affect the actual job, so they should not be really part of the output name. I will filter out them out and push to your PR before merging it.